src/jit/morph.cpp

```diff
@@ -8973,6 +8972,7 @@ GenTreePtr Compiler::fgMorphOneAsgBlockOp(GenTreePtr tree)
                     // we don't delete it as a dead store later on.
                     unsigned lclVarNum                = lclVarTree->gtLclVarCommon.gtLclNum;
                     lvaTable[lclVarNum].lvAddrExposed = true;
+#ifndef LEGACY_BACKEND
```

Comment:
Why are you removing this call to lvaSetVarDoNotEnregister here?  Setting lvaAddrExposed implies that we also cannot enregister this variable, so we call call lvaSetVarDoNotEnregister with a reason and it sets lvDoNotEnregister